app/queries/servers/system.ts

```diff
@@ -220,3 +222,10 @@ export const setCurrentTeamAndChannelId = async (operator: ServerDataOperator, t
         return {error};
     }
 };
+
+export function queryRecentMentionIds(serverDatabase: Database) {
+    return serverDatabase.get<SystemModel>(SYSTEM).findAndObserve(SYSTEM_IDENTIFIERS.RECENT_MENTIONS).pipe(
+        switchMap((row) => of$(row.value)),
+        catchError(() => of$([])),
+    );
```

Comment:
this suppose to return the result of a query (as a Promise) but you are observing here, is there a reason why you observing this instead of observing in the component that has a tear down mechanism ?

Reply:
I was confused at first on where to put what, now it clearer to me! 
I meant to delete those functions in the `queries` folder and forgot, will do now